quill-core/src/main/scala/io/getquill/norm/OrderTerms.scala

```diff
@@ -19,10 +14,25 @@ object OrderTerms {
       case Filter(SortBy(a, b, c, d), e, f) =>
         Some(SortBy(Filter(a, e, f), b, c, d))
 
-      // a.map(b => c).take(d) =>
-      //    a.take(d).map(b => c)
-      case Take(Map(a, b, c), d) =>
-        Some(Map(Take(a, d), b, c))
+      // a.flatMap(b => c).take(n).map(d => e) =>
+      //   a.flatMap(b => c).map(d => e).take(n)
+      case Map(Take(fm: FlatMap, n), ma, mb) =>
+        Some(Take(Map(fm, ma, mb), n))
+
+      // a.flatMap(b => c).drop(n).map(d => e) =>
+      //   a.flatMap(b => c).map(d => e).drop(n)
+      case Map(Drop(fm: FlatMap, n), ma, mb) =>
+        Some(Drop(Map(fm, ma, mb), n))
+
+      // a.flatMap(b => c).take(n).drop(m).map(d => e) =>
```

Comment:
I'm still not convinced that these transformations are necessary since the normalization is applied again if the tree changes. Does any test break if you remove them?